### PR TITLE
DTSPO-5049 Add Java Module Actions documentation

### PR DESCRIPTION
**Pull request by zc-hmcts**
### JIRA link ###
https://tools.hmcts.net/jira/browse/DTSPO-5049

### Change description ###
Add Java Module Actions documentation
Edit user documentation for setting up a connection via vpn

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+4. Scroll down and select 'Trigger JitPack Build' pressing the 'Set up this workflow' button
+5. On the far right select 'Start commit'
+6. Update the PR name, then select 'Commit new file'
+7. Setup the PR as a draft so we can add the remaining file.
+8. Pull your feature branch
+9. (Optional) Follow "custom configuration" step if your project is not using Java version 8
+
+## Publishing with Jitpack
+
+JitPack reads all the public HMCTS repositories and watches for new releases and tags. As soon as a new tag is found it will be available for download using the following format: ```com.github.User:Repo:Version```.
```
Comment (msl8r) [inline source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb:26]:
```suggestion
JitPack reads all the public HMCTS repositories and watches for new releases and tags. As soon as a new tag is found it will be available for download using the following format: ```com.github.hmcts:Repo:Version```.
```

File: source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
```
Comment (msl8r) [inline source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb:27]: I think it's worth mentioning how to check the status of a build and where to find the build log etc

List of files from build: `https://jitpack.io/com/github/hmcts/repo_name/tag/`
Example: `https://jitpack.io/com/github/hmcts/idam-health-checker/2.2.5/`

Build log: `https://jitpack.io/com/github/hmcts/repo_name/tag/build.log`
Example: `https://jitpack.io/com/github/hmcts/idam-health-checker/2.2.5/build.log`

File: source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+install:
+  - ./gradlew build publishToMavenLocal
+```
+
+## Forcing JitPack to build
+The first time you request a project, JitPack checks out the code, builds it and sends the Jar files back to you. This can sometimes mean that without editing timeout configuration in your gradle settings users will experience a timeout the first time a tag has been requested. To resolve this issue, a github workflow has been created to build it as soon as a tag is pushed.
+
+An example of that workflow is here: [jitpack_build.yml](https://github.com/hmcts/.github/blob/master/workflow-templates/jitpack_build.yml)
+
```
Comment (msl8r) [inline source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb:43]: Should the "using the github action" section go here?

File: source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+1. Go to your repository within GitHub
+2. Select the Actions tab at the top
+3. Select 'New workflow'
+4. Scroll down and select 'Trigger JitPack Build' pressing the 'Set up this workflow' button
+5. On the far right select 'Start commit'
+6. Update the PR name, then select 'Commit new file'
+7. Setup the PR as a draft so we can add the remaining file.
+8. Pull your feature branch
+9. (Optional) Follow "custom configuration" step if your project is not using Java version 8
```
Comment (msl8r) [inline source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb:22]:
```suggestion
9. (Optional) If your project requires custom configuration, follow the "custom configuration" step. The most common reasons for adding custom configuration is if your project is not using Java version 8 or you need a custom install command.
```

File: source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+```
+jdk:
+  - openjdk11
+
+install:
+  - ./gradlew build publishToMavenLocal
+```
+
+## Forcing JitPack to build
```
Comment (timja) [inline source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb:39]: is this needed at all? it's already included in the guide

File: source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+You will end up with something similar to the example below in your build.gradle files.
+
+```
+dependencies {
+    ...
+    implementation 'com.github.hmcts:sscs-common:4.2.39'
+}
+```
+Further information on JitPack can be found at [jitpack.io/docs](jitpack.io/docs).
```
Comment (timja) [inline source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb:69]: Can we have a section on multi-module builds that shows how to find the ID?
https://jitpack.io/#hmcts/java-logging/5.1.9
scroll down to subproject and click the dropdown? or just document the format?

File: source/ways-of-working/new-component/connect-via-vpn.html.md.erb [hunk 7]
```diff
@@ -0,0 +1,33 @@
+---
+title: Set up a access to an application via the VPN
+last_reviewed_on: 2021-10-15
+review_in: 6 months
+weight: 10
+---
+
+# <%= current_page.data.title %>
```
Comment (timja) [inline source/ways-of-working/new-component/connect-via-vpn.html.md.erb:8]: move to separate PR please

File: source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb (hunk 5 above)
Comment (zc-hmcts) [reply source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb:39]: just the "forcing jitpack to build" section is that?

File: source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb (hunk 2 above)
Comment (zc-hmcts) [reply source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb:22]: done

File: source/ways-of-working/common-pipeline/publishing-libraries/index.html.md.erb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+---
+title: Publishing Libraries
+last_reviewed_on: 2021-10-18
+review_in: 6 months
+weight: 2
```
Comment (timja) [inline source/ways-of-working/common-pipeline/publishing-libraries/index.html.md.erb:5]: can you see if you can adjust this so that the common pipeline comes before this in the side bar?

File: source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+```
+jdk:
+  - openjdk11
+
+install:
+  - ./gradlew build publishToMavenLocal
+```
+
+## Forcing JitPack to build
+The first time you request a project, JitPack checks out the code, builds it and sends the Jar files back to you. This can sometimes mean that without editing timeout configuration in your gradle settings users will experience a timeout the first time a tag has been requested. To resolve this issue, a github workflow has been created to build it as soon as a tag is pushed.
```
Comment (timja) [inline source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb:35]: can you use sentence per line format to make it easier to review, it will automatically get joined into a paragrah:

```suggestion
The first time you request a project, JitPack checks out the code, builds it and sends the Jar files back to you.
This can sometimes mean that without editing timeout configuration in your gradle settings users will experience a timeout the first time a tag has been requested.
To resolve this issue, a github workflow has been created to build it as soon as a tag is pushed.
```

File: source/ways-of-working/new-component/connect-via-vpn.html.md.erb (hunk 7 above)
Comment (timja) [reply source/ways-of-working/new-component/connect-via-vpn.html.md.erb:8]: can you delete this file from this PR please

File: source/ways-of-working/new-component/index.html.md.erb [hunk 10]
```diff
@@ -19,5 +19,4 @@ weight: 3
 - [Flux config](gitops-flux.html#application-config-in-flux)
 - [Production approvals](https://github.com/hmcts/cnp-jenkins-config/blob/master/environment-approvals.yml)
 - [Path-to-live](../path-to-live/)
-- [Accessing apps via the VPN](ssh-via-vpn.html)
-
+- [Accessing apps via the VPN](connect-via-vpn.html)
```
Comment (timja) [inline source/ways-of-working/new-component/index.html.md.erb:22]: revert change as it will cause a conflict

File: source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+1. Go to your repository within GitHub
+2. Select the Actions tab at the top
+3. Select 'New workflow'
+4. Scroll down and select 'Trigger JitPack Build' pressing the 'Set up this workflow' button
+5. On the far right select 'Start commit'
+6. Update the PR name, then select 'Commit new file'
+7. Setup the PR as a draft so we can add the remaining file.
+8. Pull your feature branch
+9. (Optional) If your project requires custom configuration, follow the "custom configuration" step. The most common reasons for adding custom configuration is if your project is not using Java version 8 or you need a custom install command.
```
Comment (timja) [inline source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb:49]: ordering doesn't seem right as the custom command is now above this, also I think we should remove optional and just always include it (below this)

Comment (zc-hmcts) [reply source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb]: changed order

File: source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb [hunk 12]
```diff
@@ -48,7 +30,18 @@ An example of that workflow is here: [jitpack_build.yml](https://github.com/hmct
 6. Update the PR name, then select 'Commit new file'
 7. Setup the PR as a draft so we can add the remaining file.
 8. Pull your feature branch
-9. (Optional) If your project requires custom configuration, follow the "custom configuration" step. The most common reasons for adding custom configuration is if your project is not using Java version 8 or you need a custom install command.
+9. If your project requires custom configuration, follow the "custom configuration" step. The most common reasons for adding custom configuration is if your project is not using Java version 8 or you need a custom install command.
+
+## Custom configuration:
+By default, Jitpack compiles projects using OpenJDK Java 8 and uses ``./gradlew install` for the maven plugin and ``./gradlew build publishToMavenLocal`` for the maven-publish plugin. If you use both, JitPack will default to using install. If that doesn’t work for your project, you will need to create a jitpack.yml file add custom configuration to it, like below.
```
Comment (timja) [inline source/ways-of-working/common-pipeline/publishing-libraries/java.html.md.erb:36]:
```suggestion
By default, Jitpack compiles projects using OpenJDK Java 8 and uses `./gradlew install` for the maven plugin and `./gradlew build publishToMavenLocal` for the maven-publish plugin.
If you use both, JitPack will default to using install.
In your `jitpack.yml` you can override the install command like so:
```
